system/jlib/jutil.cpp

```diff
@@ -2405,6 +2405,35 @@ jlib_decl StringBuffer& decodeUrlUseridPassword(StringBuffer& out, const char* i
     return out;
 }
 
+jlib_decl StringBuffer& encode_url(StringBuffer& out, const char* in)
+{
+    for (const char *p = in; *p; ++p)
+    {
+        switch (*p)
+        {
+            case ' ': out.append("%20"); break;
+            case '+': out.append("%2B"); break;
+            case '\r': out.append("%0D"); break;
+            case '\n': out.append("%0A"); break;
+            case '\'': out.append("%27"); break;
+            case ',': out.append("%2C"); break;
+            case ':': out.append("%3A"); break;
+            case ';': out.append("%3B"); break;
+            default:
```

Comment:
This list looks too short, see:  https://en.wikipedia.org/wiki/Percent-encoding#References